### DISCUSSION TITLE
Issue with aims.SurfaceManip.meshArea and Texture

### DISCUSSION THREAD

**Discussion by c-langlet** [brainvisa/brainvisa.github.io#108] [Q&A] [answered]
Hi,

_aims.SurfaceManip.meshArea(mesh, texture)_ seems to work when _texture_ is a _TimeTexture()_ but not when it is a _Texture()_

Example:
 
    from soma import aims
    mesh = aims.read('my_mesh.gii')
    texture = aims.read('my_time_texture.gii')
    aims.SurfaceManip.meshArea(mesh, texture)
    aims.SurfaceManip.meshArea(mesh, texture[0])

Produces error: 

    TypeError: SurfaceManip.meshArea(): arguments did not match any overloaded call:
    overload 1: too many arguments
    overload 2: argument 2 has unexpected type 'Texture_S16'

The issue is that _aims.SurfaceManip.meshArea(mesh, texture)_ only provides areas for the first texture of the TimeTexture.
I can dodge the issue by assigning _texture[index]_ to a TimeTexture and calculate the areas on it, but it is not very handy.
Is it expected behavior ?

**Comment by denisri** [comment, marked as answer]
Right, that's true, this function is not really complete. `Texture` was originally meant to be an "internal" structure barely exposed to the user, and `TimeTexture` was the "official" object that should be manipulated (only `TimeTexture` has IO, for instance), thus many mesh / texture functions operate on `TimeTexture`. But as developers are lazy some functions are only implemented for a single timestep, like here. The problem is that `meshArea()` returns a single timestep values set (a region: area map), not a list/map of such, so we cannot just change its internal implementation. We can either add a `meshArea(Mesh, Texture)`, or add another function with a different name like `meshAreas(Mesh, TimeTexture)`. The second would probably be more convenient, but would add another confusing discrepency in functions names, so I hesitate between both.

**Comment by denisri** [comment]
Hm actually `meshAreas(AimsSurface<3, Void> &, Texture)` exists, but not `meshAreas(AimsTimeSurface<3, Void> &, Texture)`, so the minimal solution is to add this one so that you can use it from python.

**Comment by denisri** [comment]
I have done so.

**Reply by c-langlet** [reply]
Thanks !
